File: example/resource.ts
import { serve } from "@hono/node-server";
import { Hono } from "hono";
import { logger } from "hono/logger";
import { hono } from "x402/server";

const app = new Hono();
const port = 4021;

app.use(
  "/joke",
  hono.paymentMiddleware(
    "$0.01",
    "0x209693Bc6afc0C5328bA36FaF03C514EF312287C",
    {
      description: "joke",
      mimeType: "text/plain",
    }
  )
);

app.use("*", logger());

app.get("/joke", (c) => {
  const jokes = [
    "Why do programmers prefer dark mode? Because light attracts bugs!",
    "Why did the developer go broke? Because he used up all his cache!",
    "What's a programmer's favorite place? The foo bar!",
    "Why do programmers always mix up Halloween and Christmas? Because Oct 31 equals Dec 25!",
    "Why did the programmer quit his job? Because he didn't get arrays!",
  ];

  const randomJoke = jokes[Math.floor(Math.random() * jokes.length)];
  return c.json({ joke: randomJoke });
});

console.log(`Resource running on port ${port}`);

serve({
  port: port,
  fetch: app.fetch,
});

File: go/examples/server/resource.go
package main

import (
	"math/big"

	x402gin "github.com/coinbase/x402/go/pkg/gin"
	"github.com/coinbase/x402/go/pkg/types"
	"github.com/gin-gonic/gin"
)

func main() {
	r := gin.Default()

	facilitatorConfig := &types.FacilitatorConfig{
		URL: "http://localhost:3000",
	}

	r.GET(
		"/joke",
		x402gin.PaymentMiddleware(
			big.NewFloat(0.0001),
			"0x209693Bc6afc0C5328bA36FaF03C514EF312287C",
			x402gin.WithFacilitatorConfig(facilitatorConfig),
			x402gin.WithResource("http://localhost:4021/joke"),
		),
		func(c *gin.Context) {
			c.JSON(200, gin.H{
				"joke": "Why do programmers prefer dark mode? Because light attracts bugs!",
			})
		},
	)

	r.Run(":4021") // Start the server on 0.0.0.0:4021 (for windows "localhost:4021")
}
